src/Catalog/Monitoring/ValidationTestCollector.cs

```diff
@@ -0,0 +1,76 @@
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test;
+using NuGet.Services.Metadata.Catalog.Persistence;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// Runs a <see cref="PackageValidationTestRunner"/> against collected packages.
+    /// </summary>
+    public class ValidationTestCollector : SortingIdVersionCollector
+    {
+        public ValidationTestCollector(Storage auditingStorage, PackageValidationTestRunner packageValidationTestRunner, Uri index, Func<HttpMessageHandler> handlerFunc = null, ILogger logger = null) : base(index, handlerFunc)
+        {
+            _auditingStorage = auditingStorage;
+            _packageValidationTestRunner = packageValidationTestRunner;
+            _logger = logger;
+        }
+        
+        public IEnumerable<PackageValidationTestResult> PackageValidationResults => new List<PackageValidationTestResult>();
+
+        protected override async Task ProcessSortedBatch(CollectorHttpClient client, KeyValuePair<Tuple<string, string>, IList<JObject>> sortedBatch, JToken context, CancellationToken cancellationToken)
+        {
+            var packageId = sortedBatch.Key.Item1;
+            var packageVersion = sortedBatch.Key.Item2;
+            var catalogEntriesJson = sortedBatch.Value;
+
+            try
+            {
+                var testResults = await _packageValidationTestRunner.Run(packageId, packageVersion, catalogEntriesJson, _auditingStorage, client, cancellationToken, _logger);
+                _logger.LogInformation("Finished testing {PackageId} {PackageVersion}", packageId, packageVersion);
+                foreach (var testResult in (TestResult[])Enum.GetValues(typeof(TestResult)))
+                {
+                    var validationsWithResult = testResults.EndpointValidationResults.SelectMany(r => r.ValidationResults).Where(r => testResult == r.Result);
+                    if (validationsWithResult.Any())
+                    {
+                        foreach (var validationTestResult in validationsWithResult)
+                        {
+                            var testResultLogString = "{PackageId} {PackageVersion}: {ValidationTest}: {TestResult}";
+                            var validationTestName = validationTestResult.ValidationTest.GetType().Name;
+                            var testResultString = testResult.ToString();
+
+                            switch (testResult)
+                            {
+                                case TestResult.Pass:
+                                    _logger.LogInformation(testResultLogString, packageId, packageVersion, validationTestName, testResultString);
+                                    break;
+                                case TestResult.Skip:
+                                    _logger.LogWarning(testResultLogString, packageId, packageVersion, validationTestName, testResultString);
```

Comment:
>LogWarning [](start = 44, length = 10)

Warning for skip seems really noisy. Won't a lot of things get skipped in practice? (any old catalog entry prior to an edit)